src/pages/[gatewayUID]/node/[nodeId]/details.tsx

```diff
@@ -33,33 +37,198 @@ interface SparrowQueryInterface extends ParsedUrlQuery {
   minutesBeforeNow?: string; // this value is a string to it can be a query param
 }
 
-type NodeDetailsData = {
-  viewModel: NodeDetailViewModel;
-  err?: string;
-};
+const NodeDetails: NextPage = () => {
+  const [viewModel, setViewModel] = useState<NodeDetailViewModel>({});
+  const [isLoading, setIsLoading] = useState(false);
+  const [err, setErr] = useState<string | undefined>(undefined);
+  const refetchInterval = 10000;
 
-const NodeDetails: NextPage<NodeDetailsData> = ({ viewModel, err }) => {
   const { TabPane } = Tabs;
   const { Option } = Select;
+  const router = useRouter();
   const { query } = useRouter();
 
   // neither of these values will ever be null because the URL path depends on them to render this page
-  const { gatewayUID, nodeId } = query as SparrowQueryInterface;
+  const { gatewayUID, nodeId, minutesBeforeNow } =
+    query as SparrowQueryInterface;
   const nodeUrl = `/${gatewayUID}/node/${nodeId}/details`;
 
-  const router = useRouter();
-  // Call this function whenever you want to
-  // refresh props!
-  const refreshData = async () => {
-    await router.replace(router.asPath);
-  };
+  const {
+    isLoading: gatewayLoading,
+    error: gatewayError,
+    data: gateway,
+  } = useGateway(gatewayUID);
+
+  const {
+    isRefetching: nodeRefetching, // for when we update the node name
+    isLoading: nodeLoading,
+    error: nodeError,
+    data: node,
+    refetch: nodeRefetch,
+  } = useNode(gatewayUID, nodeId, refetchInterval);
+
+  const {
+    isLoading: readingsLoading,
+    error: readingsError,
+    data: readings,
+    refetch: nodeReadingsRefetch,
+  } = useNodeData(gatewayUID, nodeId, minutesBeforeNow, refetchInterval);
+
+  useEffect(() => {
+    if (gateway && node && readings) {
+      const nodeModel: NodeDetailViewModel = getNodeDetailsPresentation(
```

Comment:
Should getNodeDetailsPresentation run client-side or should we give it an API endpoint so we can fetch everything for this page on one round-trip to the server? Would make this page simpler too I think. Or, maybe we just don't need a view-model for an entire page anymore, and we would want to just have a viewModel for each section of the page.

Reply:
Fair point. I'm hoping the Prisma implementation will be able to give us all of this in one fell swoop like you say